File: solutions/c/leap/1/leap.c
#include "leap.h"

bool leap_year(int year) {

    int isCentenary = year % 100;
    int divisibleByFourHundred = year % 400;
    int divisibleByFour = year % 4;
    
    if(isCentenary == 0) {
        return divisibleByFourHundred == 0 ? true : false;
    } else {
        return divisibleByFour == 0 ? true : false;
    }
}

File: solutions/java/leap/1/src/main/java/Leap.java
class Leap {

    boolean isLeapYear(int year) {

        boolean isCentenary = year % 100 == 0 ? true : false;
        
        if(isCentenary) {
            return year % 400 == 0 ? true : false;
        }

        return year % 4 == 0 ? true : false;
    }
}
